lock.rst

```diff
@@ -0,0 +1,291 @@
+.. index::
+    single: Lock
+
+Dealing with concurrency with Lock
```

Comment:
```suggestion
Dealing with concurrency with Locks
```

I find "with Lock" a bit weird, as you actually mean "with the Lock component". Looking at the titles of all other guides, I think we should either call this article "Locks" or my suggestion (to make a normal sentence).